src/api/feedDiscovery/src/lib/discover.js

```diff
@@ -0,0 +1,74 @@
+const got = require('got');
+const cheerio = require('cheerio'); // jQuery for node
+
+// Helper function to check the validity of the blog url
+function isValidUrl(blogUrl) {
+  // Reference source for regEx string
+  // https://gist.github.com/dperini/729294
+  const validUrlRegex = /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$/i;
+  return validUrlRegex.test(blogUrl);
+}
+
+// Helper function to return the feed url of a given blog url
+async function getFeedUrl(blogUrl) {
+  try {
+    const res = await got.get(blogUrl);
+    // Load the html document
+    const $ = cheerio.load(res.body);
+    // Extract the list of rss link elements
+    const rss = $('link[type="application/rss+xml"]').clone();
+    // Extract the list of atom link elements
+    const atom =
+      $('link[type="application/atom+xml"]').clone() ||
+      $('link[type="application/x.atom+xml"]').clone() ||
+      $('link[type="application/x-atom+xml"]').clone();
+    const feedList = rss.length ? rss : atom; // If rss has any content (i.e. it actually exists), assign its value to elem, otherwise assign the value of atom
+    const feedUrl = feedList[0].attribs.href;
+    return feedUrl;
+  } catch (e) {
+    throw new Error();
```

Comment:
Should `e` be fed into `Error();`?

Reply:
I still don't know what would be best to handling error. This is a work in progress.

Reply:
Curious about this too, my service does the same thing. Would love to know if a better way for this exists.

Reply:
In general, it's an anti-pattern to do this:

```js
try {
  ...
} catch(err) {
  throw new Error();
}
```

You just lost all context from the original error that could have been logged or used to decide what to do next.  If you're going simply re-throw the error, remove the try/catch (i.e., it's not doing anything) and let the caller deal with the exception.

Should we validate that this/these feed URLs are working before we continue?

Reply:
@humphd : I was thinking about it but then I realized that those feed urls will be extracted from the provider, would that mean it's already legit? So I skip it.

Reply:
We could try trusting them for now, sure.